cpython/patches/testing.patch

```diff
@@ -82,3 +82,30 @@ index 1fc4de11e1..e6c91707ae 100644
  import time
  import unittest
  import unittest.mock as mock
+diff --git a/Lib/test/test_gzip.py b/Lib/test/test_gzip.py
+--- a/Lib/test/test_gzip.py
++++ b/Lib/test/test_gzip.py
+@@ -13,6 +13,7 @@
+ from test import support
+ from test.support import _4G, bigmemtest
+ from test.support.script_helper import assert_python_ok
++import unittest
+ 
+ gzip = support.import_module('gzip')
+ 
+@@ -691,6 +692,7 @@
+     data = b'This is a simple test with gzip'
+ 
+     def test_decompress_stdin_stdout(self):
++        raise unittest.SkipTest('no subprocess')
+         with io.BytesIO() as bytes_io:
+             with gzip.GzipFile(fileobj=bytes_io, mode='wb') as gzip_file:
+                 gzip_file.write(self.data)
+@@ -727,6 +729,7 @@
+ 
+     @create_and_remove_directory(TEMPDIR)
+     def test_compress_stdin_outfile(self):
++        raise unittest.SkipTest('no subprocess')
```

Comment:
Test failure:

```
======================================================================
ERROR: test_compress_stdin_outfile (test.test_gzip.TestCommandLine)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/lib/python3.7/test/test_gzip.py", line 683, in wrapper
    return function(*args, **kwargs)
  File "/lib/python3.7/test/test_gzip.py", line 731, in test_compress_stdin_outfile
    with Popen(args, stdin=PIPE, stdout=PIPE, stderr=PIPE) as proc:
  File "/lib/python3.7/subprocess.py", line 769, in __init__
    restore_signals, start_new_session)
  File "/lib/python3.7/subprocess.py", line 1447, in _execute_child
    restore_signals, start_new_session, preexec_fn)
BlockingIOError: [Errno 11] Resource temporarily unavailable

======================================================================
ERROR: test_decompress_stdin_stdout (test.test_gzip.TestCommandLine)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/lib/python3.7/test/test_gzip.py", line 699, in test_decompress_stdin_stdout
    with Popen(args, stdin=PIPE, stdout=PIPE, stderr=PIPE) as proc:
  File "/lib/python3.7/subprocess.py", line 769, in __init__
    restore_signals, start_new_session)
  File "/lib/python3.7/subprocess.py", line 1447, in _execute_child
    restore_signals, start_new_session, preexec_fn)
BlockingIOError: [Errno 11] Resource temporarily unavailable

----------------------------------------------------------------------
```
https://pastebin.com/TifZ3vEh for the full output -- this failure appears to cause a secondary failure which might also be interesting?

These two tests attempt to `Popen()` the current Python process and do stuff with it.  Several other tests in this suite are already skipped due to trying to do this, though the skip is implicit / buried in a helper function.

Several tests in other files successfully `Popen()` Python.  So it looks to me like it is possible to launch a subprocess in pyodide?  Is the difference that this case is opening stdin as a pipe?  What's up with subprocesses and pipes?; is anyone thinking about that currently?

Reply:
subprocessing should not supported as far as I know and they should be skipped via the flags in https://github.com/iodide-project/pyodide/blob/master/test/python_tests.txt file.